### PR TITLE
[Merged by Bors] - feat(SetTheory/Ordinal/Arithmetic): the complement of a small set of ordinals is unbounded

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean [hunk 1]
```diff
@@ -2009,6 +2009,12 @@ the Burali-Forti paradox. -/
 theorem not_small_ordinal : ¬Small.{u} Ordinal.{max u v} := fun h =>
   @not_injective_of_ordinal_of_small _ h _ fun _a _b => Ordinal.lift_inj.{v, u}.1
 
+theorem Ordinal.compl_nonempty_of_small (s : Set Ordinal.{u}) [hs : Small.{u} s] :
+    Set.Nonempty sᶜ := by
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Ordinal/Arithmetic.lean:2013]: Can you show s^c is unbounded rather?

File: Mathlib/Logic/Small/Set.lean [hunk 2]
```diff
@@ -41,6 +41,9 @@ instance small_image2 (f : α → β → γ) (s : Set α) (t : Set β) [Small.{u
   rw [← Set.image_uncurry_prod]
   infer_instance
 
+theorem small_univ_iff : Small.{u} (@Set.univ α) ↔ Small.{u} α :=
```
Comment (YaelDillies) [inline Mathlib/Logic/Small/Set.lean:44]: Is the reverse implication already a lemma?

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean (hunk 1 above)
Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/Arithmetic.lean:2013]: Nice catch! I'll update the statement.

File: Mathlib/Logic/Small/Set.lean (hunk 2 above)
Comment (vihdzp) [reply Mathlib/Logic/Small/Set.lean:44]: No, but it seems worth adding as an instance.

**Review by YaelDillies — APPROVED** [review]
Nice!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by YaelDillies** [comment]
Last minute thought: Could you also prove `BddAbove s → ¬ Small.{u} sᶜ`?

**Comment by vihdzp** [comment]
This is basically just the contrapositive, isn't it? I don't think it's a very useful statement - you generally don't get much out of a type not being small (with the exceptions of `Ordinal` and `Cardinal`). The correct statement is probably `BddAbove s → ¬ BddAbove sᶜ`, which should hold in any `NoMaxOrder`.

**Comment by YaelDillies** [comment]
Yes, that would be a nice lemma to have

**Pull request closed without merging**